tests/cupy_tests/manipulation_tests/test_rearrange.py

```diff
@@ -115,3 +115,105 @@ def test_flipud_2(self, xp, dtype):
     def test_flipud_insufficient_ndim(self, xp, dtype):
         x = testing.shaped_arange((), xp, dtype)
         return xp.flipud(x)
+
+
+@testing.gpu
+class TestFlip(unittest.TestCase):
+
+    _multiprocess_can_split_ = True
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_flip_1(self, xp, dtype):
+        x = testing.shaped_arange((3,), xp, dtype)
+        return xp.flip(x, 0)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_flip_2(self, xp, dtype):
+        x = testing.shaped_arange((3, 4), xp, dtype)
+        return xp.flip(x, 1)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_flip_with_negative_axis(self, xp, dtype):
+        x = testing.shaped_arange((3, 4, 2), xp, dtype)
+        return xp.flip(x, -1)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_raises()
+    def test_flip_insufficient_ndim(self, xp, dtype):
+        x = testing.shaped_arange((), xp, dtype)
+        return xp.flip(x, 0)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_raises()
+    def test_flip_invalid_axis(self, xp, dtype):
+        x = testing.shaped_arange((3, 4), xp, dtype)
+        return xp.flip(x, 2)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_raises()
+    def test_flip_invalid_negative_axis(self, xp, dtype):
+        x = testing.shaped_arange((3, 4), xp, dtype)
+        return xp.flip(x, -3)
+
+
+@testing.gpu
+class TestRot90(unittest.TestCase):
+
+    _multiprocess_can_split_ = True
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_rot90_none(self, xp, dtype):
+        x = testing.shaped_arange((3, 4), xp, dtype)
+        return xp.rot90(x, 0)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_rot90_twice(self, xp, dtype):
+        x = testing.shaped_arange((3, 4, 2), xp, dtype)
+        return xp.rot90(x, 2)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_rot90_negative(self, xp, dtype):
+        x = testing.shaped_arange((3, 4, 2), xp, dtype)
+        return xp.rot90(x, -1)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
```

Comment:
`axes` option has been introduced since numpy v1.12.0.
Could you add `@testing.with_requires` as follows?
```python
@testing.with_requires('numpy>=1.12')
@testing.for_all_dtypes()
@testing.numpy_cupy_array_equal()
```